src/bin/pg_dump/dumpall_babel_utils.c

```diff
@@ -0,0 +1,129 @@
+/*-------------------------------------------------------------------------
+ *
+ * Utility routines for babelfish objects
+ *
+ * Portions Copyright (c) 1996-2021, PostgreSQL Global Development Group
+ * Portions Copyright (c) 1994, Regents of the University of California
+ *
+ * src/bin/pg_dump/dumpall_babel_utils.c
+ *
+ *-------------------------------------------------------------------------
+ */
+#include "postgres_fe.h"
+
+#include "catalog/pg_class_d.h"
+#include "catalog/pg_proc_d.h"
+#include "catalog/pg_type_d.h"
+#include "common/logging.h"
+#include "dumpall_babel_utils.h"
+#include "fe_utils/string_utils.h"
+#include "pg_backup_db.h"
+#include "pg_backup_utils.h"
+#include "pg_backup.h"
+#include "pg_dump.h"
+#include "pqexpbuffer.h"
+
+/* Babelfish virtual database to dump */
+char *bbf_db_name = NULL;
+
+/*
+ * Returns query to fetch default Babelfish users of specified logical database.
+ * For a database DB, default roles will be: DB_dbo, DB_db_owner, DB_guest
+ * drop_query decides whether the query is to DROP the roles.
+ */
+void
+getBabelfishRolesQuery(PQExpBuffer buf, char *role_catalog, bool drop_query)
+{
+	char	*escaped_bbf_db_name;
+	bool	is_builtin_db = false;
+
+	if (!bbf_db_name)
+		return;
+
+	is_builtin_db = (pg_strcasecmp(bbf_db_name, "master") == 0 ||
+			pg_strcasecmp(bbf_db_name, "tempdb") == 0 ||
+			pg_strcasecmp(bbf_db_name, "msdb") == 0)
+			? true : false;
```

Comment:
does anything at all is required for built in db?

Reply:
Actually we don't want to dump the built users/roles as they we would already be present in the target server.

Reply:
Yeah I think there can be other ways to do this since LIMIT 0 is actually not necessary here but since LIMIT 0 has constant execution time (no performance impact as such) so it looked simpler to use it.